fl4health/clients/basic_client.py

```diff
@@ -612,13 +630,16 @@ def train_by_epochs(
                 Loss is a dictionary of one or more losses that represent the different components of the loss.
         """
         self.model.train()
-        local_step = 0
         for local_epoch in range(epochs):
+            local_step = 0  # Reset local step
```

Comment:
I should add some documentation to make this more clear. But here we assume that local_step the step number in this current round of FL for a given client. So we don't want to reset this every epoch. The reason being is that for the APFL method, we only update the alpha parameter (representing the mix of client and global models) on the first step of the first epoch. We don't want this to happen multiple times if there is more than one epoch. Here is the relevant code in [ApflClient](https://github.com/VectorInstitute/FL4Health/blob/dc574f70c88cac5eb3456168d867976684dfc5d1/fl4health/clients/apfl_client.py#L35). You should be able to back out the step for your purposes with local_step mod batches_per_epoch

Reply:
Ok I can change this back to the way it was and add a bit of clarifying documentation